src/Logger.h

```diff
@@ -14,32 +14,52 @@
 
 #include <ctime>
 #include "Storage/FileData.h"
+#include "MessageType.h"
+#include <General/NamedEnum.h>
 
 class OutputBuffer;
 
+NamedEnum(LogLevel, uint8_t, OFF, WARN, INFO, DEBUG);
+
 class Logger
 {
 public:
-	Logger() noexcept;
+	Logger(LogLevel logLvl) noexcept;
 
 	void Start(time_t time, const StringRef& file) noexcept;
 	void Stop(time_t time) noexcept;
-	void LogMessage(time_t time, const char *message) noexcept;
-	void LogMessage(time_t time, OutputBuffer *buf) noexcept;
+	void LogMessage(time_t time, const char *message, MessageType type) noexcept;
+	void LogMessage(time_t time, OutputBuffer *buf, MessageType type) noexcept;
 	void Flush(bool forced) noexcept;
 	bool IsActive() const noexcept { return logFile.IsLive(); }
 	const char *GetFileName() const noexcept { return (IsActive()) ? logFileName.c_str() : nullptr; }
+	LogLevel GetLogLevel() const noexcept { return logLevel; }
+	void SetLogLevel(LogLevel newLogLevel) noexcept;
+//	bool IsLoggingEnabledFor(const MessageType mt) const noexcept;
```

Comment:
Never comment out code without explanation. If it's code that isn't current needed, ut #if 0 to disable it, and a comment.

Reply:
Added comments and `#if 0`